das/state.go

```diff
@@ -77,14 +82,19 @@ func (s *coordinatorState) handleResult(res result) {
 
 	// update failed heights
 	for h := range res.failed {
-		failCount := 1
-		if res.job.jobType == retryJob {
-			// if job was already in retry and failed again, persist attempt count
-			failCount += s.inRetry[h]
+		// if job was already in retry and failed again, persist attempt count
+		lastRetry, ok := s.inRetry[h]
+		if ok {
 			delete(s.inRetry, h)
 		}
 
-		s.failed[h] = failCount
+		nextRetry, retryExceeded := s.retryStrategy.nextRetry(lastRetry, time.Now())
+		if retryExceeded {
+			log.Warnw("header exceeded maximum amount of sampling attempts",
+				"height", h,
+				"attempts", nextRetry.count)
+		}
```

Comment:
Why do we need to limit max amount?

Reply:
With exponential factor 4 backoff, I don't see why we should limit anything.

Reply:
Probably logic seems unclear. It is exponential backoff, but after retry reaches max backoff it will keep retrying with max backoff interval, but also log warn.